battery_monitoring.py

```diff
@@ -0,0 +1,71 @@
+
+def getpreviousitem(mylist,myitem):#Given a list and an item, return previous item in list
+    for position, item in enumerate(mylist):
+        if item == myitem:
+            # First item has no previous item
+            if position == 0:
+                return None
+            # Return previous item    
+            return mylist[position-1]
+
+def get_ranges_list(input_list):#Given a sorted list of numbers, return a list of ranges
+    new_list=list(set(input_list))
+    rangelist = []
+    inrange = False
+    for item in new_list:
+        previousitem = getpreviousitem(new_list,item)
```

Comment:
Very clear and commented code. Next level: make the code so clear that comments aren't required.
E.g., to compute the new range, make another function to accept the range and return the modified one. Write the test for such a function and then implement it.
The use of inrange is clear from the comments, but it's still possible for a new person to make mistakes while changing the code.
When you split it into functions, I feel you will realise a simpler algo.